### PR TITLE
fix: Properly mark override as introduced since 0.6.0

### PR DESCRIPTION
**Pull request by Xanewok**
This would properly fail on `address private override;`, which is a valid state variable declaration in 0.4.18 (found via Sanctuary).

### CODE REVIEW

File: crates/solidity/inputs/language/src/definition.rs [hunk 1]
```diff
@@ -2424,7 +2428,7 @@ codegen_language_macros::compile!(Language(
                             enabled = Till("0.6.0"),
                             variants = [
                                 EnumVariant(reference = ModifierInvocation),
-                                EnumVariant(reference = OverrideSpecifier),
+                                EnumVariant(reference = ConstantKeyword, enabled = Till("0.5.0")),
```
Comment (OmarTawfik) [inline crates/solidity/inputs/language/src/definition.rs:2431]: not sure if `ConstantKeyword` here is intended?

File: .changeset/cuddly-dodos-exercise.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@nomicfoundation/slang": patch
+---
+
+Fix parsing `override` identifiers before 0.6.0
```
Comment (OmarTawfik) [inline .changeset/cuddly-dodos-exercise.md:5]: My understanding is that `enabled` should not affect parsing identfiers, only `reserved` should. Not sure I understand how this is affected by this PR?

**Review by OmarTawfik — CHANGES_REQUESTED** [review]
Left a couple of questions.

File: crates/solidity/inputs/language/src/definition.rs (hunk 1 above)
Comment (Xanewok) [reply crates/solidity/inputs/language/src/definition.rs:2437]: It's not, I was stacking PRs, sorry. I've rebased and pushed a single fix commit

**Comment by Xanewok** [comment]
Rebased and changed the wording in the changeset file to better hint what's fixed and how.

File: crates/solidity/inputs/language/src/definition.rs (hunk 1 above)
Comment (OmarTawfik) [reply crates/solidity/inputs/language/src/definition.rs:2437]: I see. Thanks!
The latest version here removes `OverrideSpecifier` from `UnnamedFunctionAttribute` compeltely. Is that intended? should we add a changeset if so?

Comment (Xanewok) [reply crates/solidity/inputs/language/src/definition.rs:2437]: The unnamed function was enabled only until 0.6.0 and only from that point they introduced the override specifier, so they were never usable together. I think the current changeset of
```
Mark `override` as being a valid attribute only after 0.6.0
```

states that clearly already. WDYT?

Comment (OmarTawfik) [reply crates/solidity/inputs/language/src/definition.rs:2437]: You are right. Thank you!

**Review by OmarTawfik — APPROVED** [review]
Left a couple of question

**Pull request merged**